### PR TITLE
Remove references to SunOS and HP-UX

### PR DESCRIPTION
**Pull request by benchristel**
We don't support Greenplum on SunOS or HP-UX, so we can remove code related to those platforms.

I've made an effort not to touch files that come from postgres, to avoid future merge conflicts with upstream. Let me know if that's not the right thing to do.

I did not change gpsys1, even though it references SunOS, since that behavior is documented: https://gpdb.docs.pivotal.io/540/utility_guide/admin_utilities/gpsys.html

In addition, I'm unsure what to do about the following files (which I didn't change):

```
./src/backend/port/dynloader/hpux.c
./src/backend/port/dynloader/hpux.h
./src/backend/port/dynloader/solaris.c
./src/backend/port/dynloader/solaris.h
./src/backend/port/hpux
./src/backend/port/tas/hpux_hppa.s
./src/backend/port/tas/sunstudio_sparc.s
./src/backend/port/tas/sunstudio_x86.s
./src/include/port/atomics/arch-hppa.h
./src/include/port/atomics/generic-sunpro.h
./src/include/port/hpux.h
./src/include/port/solaris.h
./src/template/hpux
./src/template/solaris
```

Dev pipeline: https://dev.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb-dev-releng-remove-sunos-hpux

### CODE REVIEW

File: src/makefiles/Makefile.hpux [hunk 1]
```diff
@@ -1,62 +0,0 @@
-# If we don't have POSIX signals, we need to use the libBSD signal routines.
```
Comment (danielgustafsson) [inline src/makefiles/Makefile.hpux]: This is an upstream file, please don't remove this.

File: src/makefiles/Makefile.solaris [hunk 2]
```diff
@@ -1,27 +0,0 @@
-# src/makefiles/Makefile.solaris
```
Comment (danielgustafsson) [inline src/makefiles/Makefile.solaris]: This is an upstream file, please don't remove.

**Comment by danielgustafsson** [comment]
> In addition, I'm unsure what to do about the following files (which I didn't change):

These are upstream files, please just leave them to avoid merge conflicts from deletions where the gain isn't high enough.

**Comment by benchristel** [comment]
@danielgustafsson I've reverted the files that were from upstream Postgres. Can you take another look?

File: gpcontrib/gpmapreduce/Makefile [hunk 3]
```diff
@@ -4,9 +4,6 @@ include $(top_builddir)/src/Makefile.global
 
 BLD_OS := $(shell uname -s)
 BLD_OS_VER := $(shell uname -r)
-ifneq "$(findstring $(BLD_OS)-$(BLD_OS_VER),SunOS-5.8 SunOS-5.9)" ""
-BLD_CPPFLAGS_SOLARIS_OLD_VERSIONS=-D_REENTRANT
-endif
```
Comment (danielgustafsson) [inline gpcontrib/gpmapreduce/Makefile]: The two statements above should be removed as well, there are no consumers of the `BLD_OS` and `BLD_OS_VER` variables left now.

**Pull request merged**